include/real/real.hpp

```diff
@@ -620,10 +642,35 @@ namespace boost {
              */
 
             static real power(real<T> real_num, real<T> exponent){
-
+ 
                 return real(real_operation<T>(real_num._real_p, exponent._real_p, OPERATION::POWER));
             }
 
+
+            /**
+             *      SIN METHOD (INPUT IN RADIANS)
+             * @brief: Calculated sin(real_num), real_num should represent angle in radians
+             * @params: real_num: boost::real number whose sin is to be calculated.
+             * @return: returns a new boost::real which is sin(real_num)
+             * @author: Vikram Singh Chundawat
+             **/
+            static real sin(real<T> real_num){
+                static real<T> zero("0");
+                return real(real_operation<T>(real_num._real_p, zero._real_p, OPERATION::SIN));
+            }
+
+            /**
+             *      COS METHOD (INPUT IN RADIANS)
+             * @brief: Calculated cos(real_num), real_num should represent angle in radians
+             * @params: real_num: boost::real number whose cos is to be calculated.
+             * @return: returns a new boost::real which is cos(real_num)
+             * @author: Vikram Singh Chundawat
+             **/
+            static real cos(real<T> real_num){
+                static real<T> zero("0");
```

Comment:
We repeating this definition in many places, we should have a Zero that is accessible by all of them.

Reply:
I think it's time to define a literals.hpp file with all the literals. Maybe I would use a namespace like real::literals

Reply:
I have one doubt on that, doesn't that create a circular dependency?
I think for these zeroes and ones, we should define them just after the class is defined, like just after integer_number class is defined, we will make these constant variables(like zero_exact_number) in real::literals namespace.
And literals.hpp will be used to store those scientific constants I proposed to add, in the same namespace (real::literals).
What I am proposing is, we do not define these variables in separate files, although we will define them in literals namespace, but not in literals.hpp, to avoid any possible circular dependencies.

Reply:
It makes sense to define them in the same file, just not defining them many many times.